go-controller/pkg/crd/egressfirewall/v1/types.go

```diff
@@ -0,0 +1,67 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EgressNetworkFirewallRuleType indicates whether an EgressNetworkFirewallRule allows or denies traffic
+type EgressFirewallRuleType string
+
+const (
+	EgressFirewallRuleAllow EgressFirewallRuleType = "Allow"
+	EgressFirewallRuleDeny  EgressFirewallRuleType = "Deny"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=egressfirewall
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// EgressFirewall describes the current egress firewall for a Namespace.
+// Traffic from a pod to an IP address outside the cluster will be checked against
+// each EgressFirewallRule in the pod's namespace's EgressFirewall, in
+// order. If no rule matches (or no EgressFirewall is present) then the traffic
+// will be allowed by default.
+type EgressFirewall struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata, omitempty"`
+	// Specification of the desired behavior of EgressFirewall.
+	Spec EgressFirewallSpec `json:"spec"`
```

Comment:
Why is there no Status? One of the mandates we have is that any new CRDs we add should conform to having `Spec` and `Status`.

Reply:
added a status field